examples/2d_isometry/tiles.rs

```diff
@@ -0,0 +1,842 @@
+// TODO: Have something to import TextureAtlas as an asset.
+// This was generated by a simple parser built in 10 minutes,
+// but it's far from being convenient.
+
+use amethyst::assets::{AssetStorage, Loader};
+use amethyst::ecs::World;
+use amethyst::renderer::{
+    MaterialTextureSet, PngFormat, Sprite, SpriteSheet, SpriteSheetHandle, SpriteSheetSet, Texture,
+    TextureCoordinates, TextureMetadata, FilterMethod, SamplerInfo, WrapMode
+};
+
+const TEXTURE_ID: u64 = 0;
+const SPRITESHEET_SIZE: (f32, f32) = (1024.0, 2048.0);
+
+struct SpriteDescription {
+    top: f32,
+    bottom: f32,
+    left: f32,
+    right: f32,
+}
+
+pub fn load_sprite_sheet(world: &mut World) -> SpriteSheetHandle {
+    let texture_handle = {
+        let loader = world.read_resource::<Loader>();
+        let texture_storage = world.read_resource::<AssetStorage<Texture>>();
+        loader.load(
+            "texture/isometric_tiles.png",
+            PngFormat,
+            TextureMetadata::default()
+                .with_sampler(SamplerInfo::new(FilterMethod::Trilinear, WrapMode::Border)),
+            (),
+            &texture_storage,
+        )
+    };
+
+    let mut material_texture_set = world.write_resource::<MaterialTextureSet>();
+    material_texture_set.insert(TEXTURE_ID, texture_handle);
+
+    let mut sprites = Vec::new();
+    for i in 0..128 {
+        let dims = get_sprite(i);
+        let tex_coords = TextureCoordinates {
+            left: (dims.left) / SPRITESHEET_SIZE.0,
+            right: (dims.right) / SPRITESHEET_SIZE.0,
+            bottom: 1.0 - (dims.bottom) / SPRITESHEET_SIZE.1,
+            top: 1.0 - (dims.top) / SPRITESHEET_SIZE.1,
+        };
+        sprites.push(Sprite {
+            width: dims.right - dims.left,
+            height: dims.bottom - dims.top,
+            offsets: [0.0, 0.0],
+            tex_coords,
+        });
+    }
+
+    let sprite_sheet = SpriteSheet {
+        texture_id: TEXTURE_ID,
+        sprites: sprites,
+    };
+
+    let sprite_sheet_handle = {
+        let loader = world.read_resource::<Loader>();
+        let sprite_sheet_store = world.read_resource::<AssetStorage<SpriteSheet>>();
+        loader.load_from_data(sprite_sheet, (), &sprite_sheet_store)
+    };
+
+    sprite_sheet_handle
+}
+
+fn get_sprite(tile_type: u8) -> SpriteDescription {
```

Comment:
Surely this can be calculated instead of being hard-coded, right?

Reply:
No, look at the spritesheet, it's a mess.

Reply:
Oh god, you're right. I wonder what the author was thinking xD

Reply:
Well it doesn't waste space I guess :thonking:

Reply:
prefab file please.